src/signer/server.rs

```diff
@@ -31,27 +38,33 @@ impl Signer for Ed25519Signer {
         &self,
         request: Request<SignWithKeyRequest>,
     ) -> Result<Response<SignWithKeyResponse>, Status> {
-
         debug!("Got Request: {:?}", request);
 
         let r = request.get_ref();
-        let matched_key = match self.keypairs.iter().find(
-            |pair| pair.pubkey == r.pub_key
-        ) {
+        let keys_guard = self.keypairs.lock().await;
```

Comment:
Consider making `keys_guard` to fall out of scope sooner to release lock.

Reply:
I just remember, the way the code is written now it can't be brought out of scope sooner!!

But I am now refactoring it a lot because of the error problem... currently there is an unbelievable long function which is impossible to work with